server/models/Trust.js

```diff
@@ -141,23 +141,36 @@ class Trust {
     if (
       originatorWallet.id !== actorWallet.id &&
       originatorWallet.id !== targetWallet.id &&
-      hasControlOverActor &&
-      hasControlOverTarget
+      origHasControlOverActor &&
+      origHasControlOverTarget
     ) {
       throw new HttpError(
         409,
         'Cannot send trust relationship request to a sub wallet with the same parent',
       );
     }
 
+    // check if actor can control the target
+    const actorHasControlOverTarget = await walletModel.hasControlOver(
+      actorWallet.id,
+      targetWallet.id,
+    );
+
     // originating wallet doesn't need to send requests to a sub wallet it manages
-    if (hasControlOverTarget) {
+    if (actorHasControlOverTarget) {
```

Comment:
Why was this replaced with actorHasControlOverTarget?

Reply:
O = Originator, A = Actor, T = Target
 
By the time the code reaches here, the check is made that O != A && O != T && O has control over both A and T. Meaning that A and T are subwallets of O, and an error displays if it is true.

Next a check is made that A has control over T. This is to deal with the case if A = O, and has control over T. Meaning T (requestee)  is already managed by A (requester). 

Next, the last check is made that T = O. Meaning that A is a subwallet of O, and  target is the originating wallet.  Meaning T (requestee) already manages A (requester).

@Kpoke The checks are mainly for displaying specific and appropriate error messages.

Reply:
Okay noted. That's fine. I just wanted to confirm

Reply:
Can you pull the latest changes I made to your PR? It'll enable the PR github action run

Reply:
@Kpoke Done, thank you